conans/util/progress_bar.py

```diff
@@ -8,6 +8,35 @@
 
 TIMEOUT_BEAT_SECONDS = 30
 TIMEOUT_BEAT_CHARACTER = '.'
+LEFT_JUSTIFY_DESC = 28
+LEFT_JUSTIFY_MESSAGE = 90
+
+
+class Slots(object):
```

Comment:
I was first concerned that this is not thread-safe. I was adding locks, but it seems that this is not really necessary. This is only if we want to preserve certain positions, which is not the case. It seems this can be completely removed.

Reply:
As far as I know, tqdm will not take care of the bar positions if we don't, so I think that the output will get messy if we don't handle this. Also, I think that the operations in the slot handling part should be thread-safe.

Reply:
I have been checking the source, code, it seems that tqdm manages the bars, I have been running a few tests. It is just the order of the bars, which might be not exactly the order you want, but I think we don't really need a predetermined order, in fact what you have implemented looks like the internal management of positions in tqdm.

Reply:
Great! I checked some time ago and removing the slot manager messed everything up but looks like it works now maybe for another reason 👍